include/protozero/byteswap.hpp

```diff
@@ -12,8 +12,19 @@ documentation.
 
 #include <cassert>
 
+// Windows is only available for little endian architectures
+// http://stackoverflow.com/questions/6449468/can-i-safely-assume-that-windows-installations-will-always-be-little-endian
+#if !defined(_WIN32) && !defined(__APPLE__)
+# include <endian.h>
+#else
+# define __LITTLE_ENDIAN 1234
+# define __BYTE_ORDER __LITTLE_ENDIAN
+#endif
+
 namespace protozero {
 
+#if __BYTE_ORDER == __LITTLE_ENDIAN
+
 template <int N>
 inline void byteswap(const char* /*data*/, char* /*result*/) {
     assert(false);
```

Comment:
Can this be a `static_assert`, such that we get an error at compile time instead of at runtime when invoking the `byteswap<X>` function template with `X` not in `[1, 4, 8]` (i.e. no explicit function template specialization)?